src/services/side-panels/side-panels-manager.tsx

```diff
@@ -0,0 +1,184 @@
+    this.logger = logger;
+  }
+
+  public addItem(item: ISidePanelItemDto): number {
+    if (SidePanelsManager.validateItem(item)) {
+      const newPanelItem: SidePanelItemDto = new SidePanelItemDto(item);
+      const {componentRef, removeComponentFunc} = this.injectPanelComponent(item);
+      const newItemWrapper: ItemWrapper = new ItemWrapper(newPanelItem, componentRef, removeComponentFunc);
+      if (item.renderIcon) this.injectIconComponent(newItemWrapper);
+      this.componentsRegistry.set(newItemWrapper.id, newItemWrapper);
+      this.logger.debug('1234 New Panel Item Added', item);
+      return newItemWrapper.id;
+    }
+    this.logger.warn('invalid SidePanelItem parameters', item);
+    throw new Error('invalid SidePanelItem parameters');
+  }
+
+  public removeItem(itemId: number): void {
+    const item: ItemWrapper | undefined = this.componentsRegistry.get(itemId);
+    if (item) {
+      this.deactivateItem(itemId);
+      item.removeComponentFunc();
+      this.componentsRegistry.delete(itemId);
+    }
+  }
+
+  public activateItem(itemId: number): void {
+    const itemMetadata: ItemWrapper | undefined = this.componentsRegistry.get(itemId);
+    if (itemMetadata) {
+      const {position, expandMode} = itemMetadata.item;
+      // Trying to activate an already active item
+      if (this.isItemActive(itemId)) return;
+      // Switch between items if currently there is an active one (without collapsing / expanding PS)
+      if (this.activePanels[position] !== null) {
+        this.deactivateItem(this.activePanels[position]!.id);
+      }
+      // Deactivate the opposite panel if is active
+      const oppositePosition: SidePanelPosition = SidePanelsManager.getOppositePanelPosition(position);
+      if (this.activePanels[oppositePosition]) {
+        this.deactivateItem(this.activePanels[oppositePosition]!.id);
+      }
+      // Update new item as active
+      itemMetadata.componentRef.current?.toggle();
+      this.expand(position, expandMode);
+      this.activePanels[position] = itemMetadata;
+      itemMetadata.item.onActivate?.();
+    }
+  }
+
+  public deactivateItem(itemId: number): void {
+    const itemMetadata: ItemWrapper | undefined = this.componentsRegistry.get(itemId);
+    if (itemMetadata) {
+      const {position} = itemMetadata.item;
+      this.activePanels[position]?.componentRef.current?.toggle();
+      this.collapse(position);
+      this.activePanels[position] = null;
+      itemMetadata.item.onDeactivate?.();
+    }
+  }
+
+  public isItemActive(itemId: number): boolean {
+    const itemMetadata: ItemWrapper | undefined = this.componentsRegistry.get(itemId);
+    return itemMetadata ? this.activePanels[itemMetadata.item.position]?.id === itemId : false;
+  }
+
+  public reset(): void {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    for (const value of this.componentsRegistry.values()) {
+      this.deactivateItem(value.id);
```

Comment:
I think in change media the items should be totally removed

Reply:
you mean to the ui components ? call to removeComponent ?